src/bit_encoding/bititer.rs

```diff
@@ -103,6 +105,35 @@ impl<I: Iterator<Item = u8>> Iterator for BitIter<I> {
     }
 }
 
+impl<'a> BitIter<std::iter::Copied<std::slice::Iter<'a, u8>>> {
+    /// Creates a new bitwise iterator from a bytewise one. Equivalent
+    /// to using `From`
+    pub fn byte_slice_window(sl: &'a [u8], start: usize, end: usize) -> Self {
+        assert!(start <= end);
+        assert!(end <= sl.len() * 8);
+
+        let actual_sl = &sl[start / 8..(end + 7) / 8];
+        let mut iter = actual_sl.iter().copied();
+
+        let read_bits = start % 8;
+        if read_bits == 0 {
+            BitIter {
+                iter,
+                cached_byte: 0,
+                read_bits: 8,
```

Comment:
29a2250ed158a237204f8c06eb9ddf560b989f46: Possibly doc: `// Force iterator to read next byte for next bit`